### PR TITLE
 Introduce JSON-B and JSON-P extensions

### LINKED ISSUE #1480: Allow to use JSON-P without RESTEasy
We had issues to use JSON-P due to missing reflection configs. As I learned from @gsmet, the required config is contributed through the RESTEasy module, while our service doesn't use that (using JSON-P directly to deserialize JSON payloads received from Kafka).

### PR DESCRIPTION
**Pull request by gsmet**
I haven't added a `FeatureBuildItem` as I don't think they are worth an
item in the feature list at startup.

Fixes #1480 and #1482 and a few other issues in passing.

Also moved to the new Jakarta artifacts.

### CODE REVIEW

**Comment by kenfinnigan** [comment]
Is it confusing that this would bring in some Jakarta EE API dependencies while everything else is still Java EE?

As there is no "official" Jakarta EE release yet, I'm not sure if we should be using the APIs yet

**Comment by gsmet** [comment]
We already use part of Jakarta and I did that mostly because yasson, the implementation, moved to the Jakarta API. I needed exclusions to avoid having the same dependency twice so better use the ones used by the latest versions.

See https://github.com/quarkusio/quarkus/blob/master/build-parent/pom.xml#L652 for activation for instance.

**Comment by kenfinnigan** [comment]
I wasn't aware of that.

At any rate, it's still confusing to use Java EE in some places and Jakarta EE in others. Combined with the fact there is no Jakarta EE platform release at present.

**Comment by gsmet** [comment]
Well, yasson moved to Jakarta artifacts so I think it makes sense to be consistent with what it requires and use them.

If/when HV moves to the Jakarta-released artifact, I will also change BV to use the Jakarta artifact.

It's not as if they were double-releasing these artifacts: the new versions have been moved to Jakarta artifacts.

**Comment by kenfinnigan** [comment]
Sure, but right now there is not a Jakarta EE platform release using those specifications.

It means we're not aligning with a Java EE or Jakarta EE platform version, but a hodge podge of specification versions

**Comment by gsmet** [comment]
I don't think that's a real issue for now. We are slowly moving to Jakarta EE, one library at a time, as soon as they move to Jakarta EE.

We don't plan to release a Java EE 8 compatible implementation, AFAIK.

@stuartwdouglas what's your take on this?

**Comment by kenfinnigan** [comment]
I don't believe we're going to release anything that's completely compatible with Java EE 8 or Jakarta EE 8.

My concern is that using a mixing of APIs from the two platforms makes it impossible for a user to know which API packages to use in their application, as it becomes entirely dependent on the extensions that are added.

A user may need APIs from Java EE 8 and Jakarta from different extensions, which is very confusing.

**Comment by Sanne** [comment]
good points @kenfinnigan . Please raise this to the mailing list?

**Comment by gsmet** [comment]
BTW, I don't think it's a good reason to hold this PR as it fixes a whole bunch of issues.

I can adjust later the exclusions/artifacts, depending on what we decide.

The discussion is not exclusively related to this PR as we already use Jakarta artifacts (activation).

**Comment by kenfinnigan** [comment]
Posted: https://groups.google.com/forum/#!topic/quarkus-dev/OALN4TVmkbk

File: extensions/resteasy-jsonb/deployment/src/main/java/io/quarkus/resteasy/jsonb/deployment/ResteasyJsonbProcessor.java [hunk 1]
```diff
@@ -15,23 +15,14 @@
...
 
-import org.eclipse.yasson.JsonBindingProvider;
-import org.glassfish.json.JsonProviderImpl;
-
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
 
 public class ResteasyJsonbProcessor {
```
Comment (cescoffier) [inline extensions/resteasy-jsonb/deployment/src/main/java/io/quarkus/resteasy/jsonb/deployment/ResteasyJsonbProcessor.java]: Do we have to keep this one?

**Review by cescoffier — APPROVED** [review]

File: extensions/resteasy-jsonb/deployment/src/main/java/io/quarkus/resteasy/jsonb/deployment/ResteasyJsonbProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/resteasy-jsonb/deployment/src/main/java/io/quarkus/resteasy/jsonb/deployment/ResteasyJsonbProcessor.java]: Yes, I think it's better to have it as it simplifies using RESTEasy + JSON which is a **very** common use case.

**Comment by gsmet** [comment]
@cescoffier @kenfinnigan I changed the dependencies to use the Java EE ones for JSON-B and JSON-P.

I didn't change the Activation one for now, which was preexisting as we are a bit too close of the release.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**